### PR TITLE
New package: Insco.FaForever version 0.0.1.23

### PR DESCRIPTION
**Pull request by kjxbyz**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/159138)

### CODE REVIEW

**Comment by kjxbyz** [comment]
@microsoft-github-policy-service agree

**Comment by kjxbyz** [comment]
The issue has been solved. Please help to rerun the failed jobs.

**Comment by kjxbyz** [comment]
@stephengillie What does this check called `WinGetSvc-Validation (WinGetSvc_Validation Installer Validation)` do and why does it take so long?

**Comment by kjxbyz** [comment]
Workflow check is crashing me, fails after running for hours.

**Comment by kjxbyz** [comment]
@microsoft-github-policy-service Please explain

**Comment by kjxbyz** [comment]
Is there something wrong with the manifest? The check keeps failing!

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335187
> 2024-06-21 15:17:06.187 [FAIL] Installer failed security check. Url: https://github.com/cyf/faforever/releases/download/v0.0.0+22/FaForever_0.0.0.22_x86_en-US.msi Result: 0x80004005

Automated error analysis suggests -1978335187 may mean Installer failed security check. Another script should retry this every 18 hours. 

(Automated response - build 886.)

**Comment by stephengillie** [comment]
Hi @kjxbyz,

Several things are happening:
- This package was being impacted by a pipeline issue. https://github.com/microsoft/winget-pkgs/discussions/155218
- After clearing the pipeline issue, the package didn't pass a Defender scan. This isn't uncommon, and the majority pass in a day or a few. 
- Unfortunately, this is something of a normal route for PRs traversing our pipelines. Please don't feel especially impacted.

**Comment by kjxbyz** [comment]
@stephengillie Defender scan failed, is it because there is no code signing?

**Comment by kjxbyz** [comment]
I really can't find out what caused it? Because CI doesn't provide detailed error information. The certificate I'm using now was generated through the command line, I'm not sure if it's a problem with this certificate?

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335187
> 2024-06-22 13:22:06.071 [FAIL] Installer failed security check. Url: https://github.com/cyf/faforever/releases/download/v0.0.1+23/FaForever_0.0.1.23_x64_en-US.msi Result: 0x80004005

Automated error analysis suggests -1978335187 may mean Installer failed security check. Another script should retry this every 18 hours. 

(Automated response - build 886.)

**Comment by stephengillie** [comment]
> I really can't find out what caused it? Because CI doesn't provide detailed error information. The certificate I'm using now was generated through the command line, I'm not sure if it's a problem with this certificate?

Hi @kjxbyz,

Very sorry to let this PR sit for so long. This pacakge didn't pass a Defender scan, but hopefully it was just a false positive. I think the installer was stopped before it could interact with the certificate. 

The CI *does* provide detailed error information...

> It's a bit of a quest:
> - Click the [Checks](https://github.com/microsoft/winget-pkgs/pull/157658/checks?check_run_id=26176504766) tab.
> - Unfurl `Azure Pipelines`, click `WinGetSvc-Validation` to open that section, then click [View more details on Azure Pipelines](https://dev.azure.com/ms/ed6a5dfa-6e7f-413b-842c-8305dd9e89e6/_build/results?buildId=581650).
> - In the tab that just opened, near `Related`, click [4 published; 1 consumed](https://dev.azure.com/ms/winget-pkgs/_build/results?buildId=581650&view=artifacts&pathAsName=false). (The number of published might vary a bit.)
> - Find `InstallationVerificationLogs` and follow that line to the right edge. Put your mouse there, and 3 vertical dots should appear. Click them, then click `Download artifacts`. This should start a download in your browser.
> - Unzip the download - it should contain a set of text files and logs foreach combination of architecture and scope that was validated. 
> 
> I've automated the above into a PowerShell script, which then filters for "error", "exception" and "fail", and then filters out the noise. Then matches any strings against a list of error codes, and adds these "automated error analysis" bits to the output string, and posts all of this to the PR comments thread.

**Comment by stephengillie** [comment]
Verify:
image
image
image

Hi @kjxbyz,

Since the application shows `0.0.1+23`, but the installer writes `0.0.1.23` to the Registry, should the `PackageVersion` be updated to match the application, and a `DisplayVersion: 0.0.1.23` added, under an `AppsAndFeaturesEntries` line? Otherwise, this PR appears ready to merge.

**Comment by kjxbyz** [comment]
@stephengillie This is the automatic update mechanism in the application. After submitting the PR again, I have released two more versions, and I will disable this function later.

File: manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.locale.en-US.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Created with YamlCreate.ps1 v2.4.1 $debug=NVS1.CRLF.7-4-2.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.6.0.schema.json
+
+PackageIdentifier: Insco.FaForever
+PackageVersion: 0.0.1.23
+PackageLocale: en-US
+Publisher: Qingdao INSCO Network Technology Co., Ltd.
```
Comment (Trenly) [inline manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.locale.en-US.yaml:7]:
```suggestion
Publisher: kjxbyz
```
This needs to match control panel exactly

File: manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.locale.zh-CN.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Created with YamlCreate.ps1 v2.4.1 $debug=NVS1.CRLF.7-4-2.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.6.0.schema.json
+
+PackageIdentifier: Insco.FaForever
+PackageVersion: 0.0.1.23
+PackageLocale: zh-CN
+Publisher: Qingdao INSCO Network Technology Co., Ltd.
```
Comment (Trenly) [inline manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.locale.zh-CN.yaml:7]:
```suggestion
Publisher: kjxbyz
```

File: manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- silentWithProgress
+AppsAndFeaturesEntries:
+- DisplayVersion: 0.0.1
+UpgradeBehavior: install
+ReleaseDate: 2024-06-17
+Installers:
+- Architecture: x64
+  InstallerUrl: https://github.com/cyf/faforever/releases/download/v0.0.1+23/FaForever_0.0.1.23_x64_en-US.msi
+  InstallerSha256: 7D452D0552A4A891AB1BFBF69953E725E4FCF032665EA897C139BFE60F35F4BF
+  ProductCode: '{4588C71C-9148-41B7-8327-37676105D419}'
```
Comment (Trenly) [inline manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.installer.yaml:23]: These product codes don't look correct
```
--> Comparing ARP Entries

DisplayName                     DisplayVersion Publisher             ProductCode                            Scope
-----------                     -------------- ---------             -----------                            -----
FaForever                       0.0.1.23       kjxbyz                {05286340-8BD3-4C3F-B298-A5FCFF5A1655} Machine
```

**Review by Trenly — CHANGES_REQUESTED** [review]

File: manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.installer.yaml (hunk 3 above)
Comment (stephengillie) [reply manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.installer.yaml:23]: How are these being extracted from the Registry?

File: manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.locale.en-US.yaml (hunk 1 above)
Comment (kjxbyz) [reply manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.locale.en-US.yaml:7]: What is the control panel?

File: manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.installer.yaml (hunk 3 above)
Comment (kjxbyz) [reply manifests/i/Insco/FaForever/0.0.1.23/Insco.FaForever.installer.yaml:23]: How do I get the product code correctly?

**Comment by kjxbyz** [comment]
Check failed again 😂

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978335187
> 2024-06-29 15:21:05.987 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerSharedLib\SHA256.cpp(144)\WindowsPackageManager.dll!00007FFD2A8D273F: (caller: 00007FFD2A91AEC4) Exception(1) tid(aa8) 8A150040 
> 2024-06-29 15:21:07.933 [FAIL] Installer failed security check. Url: https://github.com/cyf/faforever/releases/download/v0.0.1+23/FaForever_0.0.1.23_x64_en-US.msi Result: 0x80004005

Automated error analysis suggests -1978335187 may mean Installer failed security check. Another script should retry this every 18 hours. 

(Automated response - build 886.)

**Comment by stephengillie** [comment]
Hi @kjxbyz,

Unfortunately, it seems like there's some kind of network block preventing our pipelines from downloading the package. There are several scans performed before installing into a VM, and these have to be performed by the pipeline. (If it gets to the second stage - install in VM - then I can manually complete from there. But this is not far enough along for me to manually complete.)

I wish I had more information on what kind of blocking, or how it was detected. If you have a business or social relationship with the developer, you might be able to find out more.

**Pull request closed without merging**